river/tree/_nodes/base.py

```diff
@@ -50,6 +50,24 @@ def filter_instance_to_leaf(self, x: dict, parent: 'Node', parent_branch: int) -
         """
         return FoundNode(self, parent, parent_branch)
 
+    def path(self, x) -> Iterator['Node']:
```

Comment:
Interesting. Did you notice any difference in the training/testing time due to this new way to traverse the tree?

Reply:
I did not run any benchmarks per se. This is the way creme traverses a path in the tree. In skmultiflow, we retrieve the leaf rather than using a generator. Perhaps @MaxHalford can shed some light on that matter. 

We might consider changing the strategy in the future. Note that in river we update only the statistics of the leaves upon sorting a new instance down the tree, whereas creme updates all the nodes in the path.